spec/models/service_template_spec.rb

```diff
@@ -355,19 +355,41 @@
       expect(sub_svc).to include(@svc_d)
     end
 
-    it "should add_resource! only if a parent_svc exists" do
-      sub_svc = instance_double("service_task", :options => {:dialog => {}}, :get_user => service_user)
-      parent_svc = instance_double("service_task", :options => {:dialog => {}})
-      expect(parent_svc).to receive(:add_resource!).once
+    describe "#create_service" do
+      let(:service_task) do
+        FactoryGirl.create(:service_template_provision_task,
+                           :miq_request => service_template_request,
+                           :options     => {:service_resource_id => service_resource.id})
+      end
+      let(:service_template_request) { FactoryGirl.create(:service_template_provision_request, :requester => user) }
+      let(:service_resource) do
+        FactoryGirl.create(:service_resource,
+                           :resource_type => 'MiqRequest',
+                           :resource_id   => service_template_request.id)
+      end
+      let(:user) { FactoryGirl.create(:user) }
+      let(:parent_service) { FactoryGirl.create(:service) }
 
-      @svc_a.create_service(sub_svc, parent_svc)
-    end
+      it "create service sets parent service resource resource id" do
+        @svc_a.create_service(service_task, parent_service)
+
+        expect(parent_service.service_resources.first.resource_id).to eq(parent_service.children.first.id)
```

Comment:
Checking the id alone is not good enough.  We should validate that it is referencing the correct object.

The `expect` could change to:
`expect(parent_service.service_resources.first.resource).to eq(parent_service.children.first)`

But that passes since the object is there in-memory so you would need to add `parent_service.reload` before checking.